src/datasets/dataset_dict.py

```diff
@@ -1417,6 +1423,57 @@ def push_to_hub(
             revision=branch,
         )
 
+    def to_pandas(
+        self,
+        splits: Optional[Union[Literal["all"], List[str]]] = None,
+        batch_size: Optional[int] = None,
+        batched: bool = False,
+    ) -> Union[pd.DataFrame, Iterator[pd.DataFrame]]:
+        """Returns the dataset as a :class:`pandas.DataFrame`. Can also return a generator for large datasets.
+
+        You must specify which splits to convert if the dataset is made of multiple splits.
+
+        Args:
+            splits (:obj:`List[str]`, optional): List of splits to convert to a DataFrame.
```

Comment:
We should update the type hints here accordingly to the ones in the signature of the method.